modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java

```diff
@@ -287,6 +336,14 @@ private VisorValidateIndexesJobResult call0() {
                     idxResults.putAll(idxRes);
             }
 
+            for (; curCacheSize < cacheSizeFutures.size(); curCacheSize++)
+                cacheSizeFutures.get(curCacheSize).get3().get();
+
+            for (; curIdxSize < idxSizeFutures.size(); curIdxSize++)
+                idxSizeFutures.get(curIdxSize).get3().get();
```

Comment:
What will happen if I just remove this waiting of futures?
I sure it will be in the method checkSizes.

Reply:
I thought it correct, because concurrent structures (cacheSizeByTbl, failCalcCacheSizeGrpIds) should be filled before chechSize() method invoked. Although you can to think about calculating index size asynchronously.